### PR TITLE
Do not base asyncTaskQueue spawn arguments directly on Sys.argv

### PR DESCRIPTION
**Pull request by SkySkimmer**
Instead we generate the needed arguments from the parsed Coqargs and stm args.

As workers do not do certain operations like Require and get their state from the parent process's summary this turns out to be very few arguments (unless I made a mistake).

Overlays:
- https://github.com/ejgallego/coq-serapi/pull/428
- https://github.com/coq/vscoq/pull/931

### CODE REVIEW

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Comment by gares** [comment]
The approach makes a lot of sense.
Some options that may not be in the initial summary are debug ones, like `-bt` (but I did not check).

**Comment by SkySkimmer** [comment]
-bt (as part of the Debug option) is in the summary but we may still want to pass it so that we get backtraces before the summary is installed.

File: stm/stmargs.ml [hunk 1]
```diff
@@ -47,7 +47,42 @@ let get_cache opt = function
...
+
+     Flags.output_directory
+     native output dir, native includes
+
+     Coqargs whose effect is used before the summary is installed and matter for workers:
+
+     -I, -boot
+
+     and we force -q -noinit otherwise the worker will try to require the
+     rcfile and prelude before receiving the summary
```
[the remark this answers is not in the document]
Comment (SkySkimmer) [reply stm/stmargs.ml:61]: also missing test-mode

**Comment by SkySkimmer** [comment]
@gares this should be ready

**Review by gares — APPROVED** [review]

**Comment by gares** [comment]
@coqbot merge now

**Pull request merged**